create resend.com template

danilowoz opened the pull request:
# Description

Creation of the Domain Connect template for resend.com to enable email signing and specify authorized senders.

## Type of change

Please mark options that are relevant.

- [x] New template
- [ ] Bug fix (non-breaking change which fixes an issue in the template)
- [ ] New feature (non-breaking change which adds functionality to the template)
- [ ] Breaking change (fix or feature that would cause existing template behavior to be not backward compatible)

# How Has This Been Tested?

Please mark the following checks done
- [x] Schema validated using JSON Schema [template.schema](./template.schema)
- [ ] Template functionality checked using [Online Editor](https://pdnsadmin.revproxy.short-lived.de/dc/free/templateedit)
- [ ] Template is checked using [template linter](https://github.com/Domain-Connect/dc-template-linter)
- [x] Template file name follows the pattern `<providerId>.<serviceId>.json`

# Example variable values

```
region: us-east-1
publicKey: RANDOM-VALUE
```

kerolasa commented:
Commit 9dd057da7f681f1762399f316ec48919ea657900 fixes typo but is titled 'create resend.com template' and the commit ca2a761de8972831427722780ce57849904f1274 has message 'fix'  that adds the template. Feels like the messages are mixed up, and to be honest it would be better to just add the template and leave typo fixing out of this review.

Secondly, it would be nice if the template would use SPFM for spf management.

```
$ dc-template-linter resend.com.mail.json
2025-01-16T16:28:14Z INF code=DCTL1014 dctl_note="use SPFM instead of bare SPF record" groupid= record=1 template=resend.com.mail.json type=TXT
```

See more info in domain connect spec: https://github.com/Domain-Connect/spec/blob/master/Domain%20Connect%20Spec%20Draft.adoc#6103-spf-record-merging

Finally the `"ttl": 0,` is not great. I advise to use 10 seconds or something short if you really want to  avoid caching.

danilowoz commented:
I totally messed up the commits, and I've dropped the commit fixing the typo to clean up the git history. Regarding the TTL, this must be a typo, and I updated it to use the default value. 

Checking the SPFM suggestion...

kerolasa commented:
The pull looks now good to me. Thanks for fixes @danilowoz
